geo/src/algorithm/minimun_rotated_rect.rs

```diff
@@ -0,0 +1,100 @@
+use crate::{
+    algorithm::{centroid::Centroid, rotate::Rotate, BoundingRect, CoordsIter},
+    Area, ConvexHull, CoordFloat, GeoFloat, GeoNum, Point, Polygon,
+};
+/// Return the minimun bounding rectangle of polygon
+///
+/// # Examples
+///
+/// ```
+/// use geo_types::{line_string, polygon, LineString, Polygon};
+/// use geo::MinimunRotatedRect;
+/// fn returns_polygon_mbr() {
+///     let poly: Polygon<f64> = polygon![(x: 3.3, y: 30.4), (x: 1.7, y: 24.6), (x: 13.4, y: 25.1), (x: 14.4, y: 31.0),(x:3.3,y:30.4)];
+///     let mbr = MinimunRotatedRect::minimun_rotated_rect(&poly).unwrap();
+///     assert_eq!(
+///         mbr.exterior(),
+///         &LineString::from(vec![
+///             (1.7000000000000006, 24.6),
+///             (1.4501458363715918, 30.446587428904767),
+///             (14.4, 31.0),
+///             (14.649854163628408, 25.153412571095235),
+///             (1.7000000000000006, 24.6),
+///         ])
+///     );
+/// }
+/// ```
+pub trait MinimunRotatedRect<'a, T> {
+    type Scalar: GeoNum;
+    fn minimun_rotated_rect(&'a self) -> Option<Polygon<Self::Scalar>>;
+}
+
+impl<'a, T, G> MinimunRotatedRect<'a, T> for G
+where
+    T: CoordFloat + GeoFloat + GeoNum,
+    G: CoordsIter<'a, Scalar = T>,
+    f64: From<T>,
+{
+    type Scalar = T;
+    fn minimun_rotated_rect(&'a self) -> Option<Polygon<Self::Scalar>> {
+        let convex_poly = ConvexHull::convex_hull(self);
+        let mut min_area = f64::MAX;
```

Comment:
Can we use the generic [`T::max_value`](https://docs.rs/num-traits/0.2.15/num_traits/float/trait.Float.html#tymethod.max_value) instead of `f64` here?